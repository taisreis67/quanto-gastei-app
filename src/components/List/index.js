import React, { Component } from 'react';
import ls from 'local-storage';
import Message from '../Message';
import Transaction from '../Transaction';

class List extends Component {
    constructor(props) {
        super(props);

        const transactions = ls.get('transactions');

        this.state = {
            transactions,
        };
    }

    message = () => {
        if (this.props.location.state) {
            const {
                textMessage,
                classMessage,
            } = this.props.location.state;

            return <Message textMessage={textMessage} classMessage={classMessage} />;
        }
    }

    listTransactions = () => {
        const {
            transactions,
        } = this.state;

        const list = transactions.map((transaction, index) => {
            return <Transaction value={transaction[0].value} description={transaction[1].value} key={index} />;
        });

        return list;
    }

    render() {
        return (
            <article>
                <header className="jumbotron jumbotron-fluid">
                    <div className="container">
                        <div className="row">
                            <div className="col">
                                <h2 className="display-4">Transações</h2>
                            </div>
                        </div>
                    </div>
                </header>

                <div className="container">
                    {this.message()}

                    <div className="row">
                        <div className="col">
                            <ul className="list-group list-group-flush">
                                {this.listTransactions()}
                            </ul>
                        </div>
                    </div>
                </div>
            </article>
        );
    }
}

export default List;