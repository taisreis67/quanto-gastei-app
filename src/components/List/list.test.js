import React from 'react';
import List from '../List';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ls from 'local-storage';

it('Deve renderizar uma mensagem de nenhuma transação cadastrada', () => {
    const list = mount(<List />);

    expect(toJson(list)).toMatchSnapshot();
});

it('Deve renderizar uma listagem com as transações adicionadas', () => {
    const transactions = [
        ['R$ 230,50', 'Tênis'],
        ['R$ 230,50', 'Tênis'],
    ];

    ls.set('transactions', transactions);

    const list = mount(<List />);

    expect(toJson(list)).toMatchSnapshot();
});