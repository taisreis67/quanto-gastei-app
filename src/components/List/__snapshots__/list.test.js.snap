// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar uma listagem com as transações adicionadas 1`] = `
<List>
  <article
    className="container"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <ul
          className="list-group list-group-flush"
        >
          <Transaction
            description="Tênis"
            key="0"
            value="R$ 230,50"
          >
            <li
              className="list-group-item"
            >
              R$ 230,50
               | 
              Tênis
            </li>
          </Transaction>
          <Transaction
            description="Tênis"
            key="1"
            value="R$ 230,50"
          >
            <li
              className="list-group-item"
            >
              R$ 230,50
               | 
              Tênis
            </li>
          </Transaction>
        </ul>
      </div>
    </div>
  </article>
</List>
`;

exports[`Deve renderizar uma mensagem de nenhuma transação cadastrada 1`] = `
<List>
  <article
    className="container"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        Nenhuma transação foi adicionada.
      </div>
    </div>
  </article>
</List>
`;
